stack build fails on macOS because of cabal's bug

## Summary
The same with https://github.com/haskell/cabal/issues/4739

## Steps to reproduce
Just execute the following on the environment mentioned below

```shell
stack build
```

environments:
```
$ stack --version
Version 2.7.1, Git revision 8afe0c2932716b0441cf4440d6942c59568b6b19 x86_64 hpack-0.34.4
```
macOS Catalina Version 10.15.5

## Expected behavior
build succeeds

## Actual behavior
```shell
$ stack build
Jikka> configure (lib + exe)
Configuring Jikka-5.0.5.0...
Jikka> build (lib + exe)
Preprocessing library for Jikka-5.0.5.0..
Building library for Jikka-5.0.5.0..
Preprocessing executable 'jikka' for Jikka-5.0.5.0..
Building executable 'jikka' for Jikka-5.0.5.0..

/Users/koki/programming/others/Jikka/<built-in>:15:10: error:
     error: non-portable path to file '".stack-work/dist/x86_64-osx/Cabal-3.2.1.0/build/Jikka/autogen/cabal_macros.h"'; specified path differs in case from file name on disk [-Werror,-Wnonportable-include-path]
#include ".stack-work/dist/x86_64-osx/Cabal-3.2.1.0/build/jikka/autogen/cabal_macros.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         ".stack-work/dist/x86_64-osx/Cabal-3.2.1.0/build/Jikka/autogen/cabal_macros.h"
1 error generated.

/Users/koki/programming/others/Jikka/.stack-work/dist/x86_64-osx/Cabal-3.2.1.0/build/jikka/autogen/Paths_Jikka.hs:1:1: error:
    `gcc' failed in phase `C pre-processor'. (Exit code: 1)
  |
1 | {-# LANGUAGE CPP #-}
  | ^


--  While building package Jikka-5.0.5.0 (scroll up to its section to see the error) using:
      /Users/koki/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_3.2.1.0_ghc-8.10.4 --builddir=.stack-work/dist/x86_64-osx/Cabal-3.2.1.0 build lib:Jikka exe:jikka --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

--- a/package.yaml
+++ b/package.yaml
@@ -39,6 +39,7 @@ library:
   ghc-options:
   - -W
   - -Werror
+  - -optP-Wno-nonportable-include-path
 
 executables:
   jikka:
@@ -50,6 +51,7 @@ executables:
     - -threaded
     - -rtsopts
     - -with-rtsopts=-N
+    - -optP-Wno-nonportable-include-path
     dependencies:
     - Jikka
 
@@ -63,6 +65,7 @@ tests:
     - -threaded
     - -rtsopts
     - -with-rtsopts=-N
+    - -optP-Wno-nonportable-include-path
     dependencies:
     - Jikka
     - ormolu
@@ -78,6 +81,7 @@ tests:
     - -threaded
     - -rtsopts
     - -with-rtsopts=-N
+    - -optP-Wno-nonportable-include-path
     dependencies:
     - Jikka
     - doctest

